chia/consensus/blockchain.py

```diff
@@ -731,6 +732,11 @@ def clean_block_records(self):
     async def get_block_records_in_range(self, start: int, stop: int) -> Dict[bytes32, BlockRecord]:
         return await self.block_store.get_block_records_in_range(start, stop)
 
+    def try_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
```

Comment:
why change this ?
it was previously handled in BlockchainInterface so to not duplicate this code here and in the wallet

Reply:
Because this is an exploration of using a `Protocol` instead of inheritance to actually check the interface.  Note that it was already duplicated into one of the blockchain classes and was the only implemented  method on `BlockchainInterface`.  If the copying is problematic we can consider various other options to avoid it.

Reply:
this was to avoid replicating the same function all in the interface wallet blockchain and cache

Reply:
Yes, and inheritance isn't the only way to share code.  I thought I already added it back the other places.  And yes, that's the quick just hack it in bit.  The more interesting part I think is how the other classes weren't actually implementing the interface (in more ways than we already knew).